### ISSUE TITLE
Error dump upload

### ISSUE THREAD

**Issue by thatoddmailbox** [CoursesPlus/CoursesPlus#11]
It would be great if there was an error telemetry feature (upload error logs to server automatically) built in (but of course could be turned off). It could hook in to the component error modal that already exists and handle unhandled exceptions in the background and option pages.

This would be useful because not many bother to email debug reports and errors are obviously a bad thing.

We could also add a "Submit debug info" to allow people to do this automatically.

Question @nywillb @medensqui do you think it should include storage data?

**Comment by willbarkoff** [comment]
Yes, but only private data.

**Comment by thatoddmailbox** [comment]
@nywillb What do you mean by private data? Isn't it all private, or am I missing something?

**Comment by willbarkoff** [comment]
@thatoddmailbox sorry, I meant public data. The report will not include private keys.

**Comment by willbarkoff** [comment]
Sorry. I closed it by accident.

**Comment by thatoddmailbox** [comment]
@nywillb Oh... you mean like schedules keys. The problem with that is that, right now, there is no way for CoursesPlus to distinguish between "public" and "private" data.

**Comment by willbarkoff** [comment]
Yes. @thatoddmailbox would it be possible to add two exports, public data
and private data?

**Comment by thatoddmailbox** [comment]
@nywillb 

> The problem with that is that, right now, there is no way for CoursesPlus to distinguish between "public" and "private" data.

**Comment by willbarkoff** [comment]
@thatoddmailbox I was talking about future versions.
